lib/src/components/mixins/draggable.dart

```diff
@@ -1,36 +1,93 @@
 import 'package:flutter/foundation.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/material.dart';
 
-import '../../gestures.dart';
-import '../base_component.dart';
-import '../../extensions/offset.dart';
+import '../../../extensions.dart';
 import '../../game/base_game.dart';
+import '../base_component.dart';
 import '../component.dart';
 
 mixin Draggable on BaseComponent {
-  bool onReceiveDrag(DragEvent event) {
+  bool onDragStarted(int pointerId, Vector2 startPosition) {
+    return true;
+  }
+
+  bool onDragUpdated(int pointerId, DragUpdateDetails details) {
+    return true;
+  }
+
+  bool onDragEnded(int pointerId, DragEndDetails details) {
+    return true;
+  }
+
+  bool onDragCanceled(int pointerId) {
+    return true;
+  }
```

Comment:
technically I could have made these void. since we check for pointers at this stage. but I decided to keep consistent, if a user wants to do something where the drag has multiple owners or change owners during the drag they can.

Reply:
But one pointer could propagate to several children still?

Reply:
it could if you like, return false on Start and then return true on Update. that is the reason I considered removing the bool from anything other the Start. but I though this was more consistent

Reply:
Right, makes sense